test/functional/feature_framework_startup_failures.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Verify framework startup failures only raise one exception since
+multiple exceptions being raised muddies the waters of what actually
+went wrong. We should maintain this bar of only raising one exception as
+long as additional maintenance and complexity is low.
+
+Test relaunches itself into child processes in order to trigger failures
+without the parent process' BitcoinTestFramework also failing.
+"""
+
+from test_framework.util import rpc_port
+from test_framework.test_framework import BitcoinTestFramework
+
+from hashlib import md5
+import re
+import subprocess
+import sys
+import time
+
+class FeatureFrameworkStartupFailures(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        # Don't start the node yet, as we want to measure during run_test.
+        self.add_nodes(self.num_nodes, self.extra_args)
+
+    # Launches a child test process running this same file, but instantiating
+    # a child test, and verifies that we (only) raise the expected exception.
+    def _verify_startup_failure(self, test, internal_args, expected_exception):
+        args = [
+            sys.executable,
+            __file__,
+            f"--cachedir={self.options.cachedir}",
+            f"--portseed={self.options.port_seed}",
+            f"--internal_test={test.__name__}",
+        ] + internal_args
+        if self.options.nocleanup:
+            args.append("--nocleanup")
+        weak_hash = md5(expected_exception.encode("utf-8"))
+        if self.options.tmpdir:
+            args.append(f"--tmpdir={self.options.tmpdir}/{weak_hash.hexdigest()[:16]}")
+        if self.options.randomseed is not None:
+            args.append(f"--randomseed={self.options.randomseed ^ int.from_bytes(weak_hash.digest()[:8], 'little')}")
```

Comment:
I'm fine with this as well, but to avoid complexity on declaration site and usage site (and to skip the "weak" part which seems out-of-place), we could try:
```suggestion
        hash_id = int(sha256(expected_exception.encode("utf-8")).hexdigest()[:8], 16)
        if self.options.tmpdir:
            args.append(f"--tmpdir={self.options.tmpdir}/{hash_id}")
        if self.options.randomseed is not None:
            args.append(f"--randomseed={self.options.randomseed ^ hash_id}")
```

Reply:
Took part of the simplification.